File: java/composite/exercise/Product.java
package composite.exercise;

public class Product implements ShoppingItem {
    private String name;
    private double price;

    public Product(String name, double price) {
        this.name = name;
        this.price = price;
    }

    @Override
    public double calculateTotal() {
        return price;
    }
}

File: C#/Composite/Exercise/Product.cs
namespace Composite.Exercise;

public sealed class Product : IShoppingItem
{
    private readonly string _name;
    private readonly double _price;

    public Product(string name, double price)
    {
        _name = name;
        _price = price;
    }

    public double CalculateTotal() => _price;
}
